src/mono/wasm/sanitize.py

```diff
@@ -14,50 +19,57 @@ def remove(*paths):
         except OSError as error:
             print(error)
 
+
 def rewrite_package_json(path):
-    package = open(path,"rb+")
+    package = open(path, "rb+")
     settings = json.load(package)
     settings["devDependencies"] = {}
     package.seek(0)
     package.truncate()
     json.dump(settings, package, indent=4)
     package.close()
 
+
 emsdk_path = sys.argv[1]
 emscripten_path = os.path.join(emsdk_path, "upstream", "emscripten")
 node_root = os.path.join(emsdk_path, "node")
 node_paths = glob(node_root)
-upgrade = False
+upgrade = True
 
-npm = os.path.join(node_paths[0], "bin", "npm")
-if not os.path.exists(npm):
-    npm = "npm"
+# Add the local node bin directory to the path so that
+# npm can find it when doing the updating or pruning
+os.environ["PATH"] += os.pathsep + os.path.join(node_paths[0], "bin")
 
 def update_npm(path):
     try:
         os.chdir(os.path.join(path, "lib"))
-        os.system(npm + " install npm@latest")
+        os.system("npm install npm@latest")
+        os.system("npm install npm@latest")
```

Comment:
Why twice?

Reply:
because the first one is run with the old npm and doesn't create a package lock, then if you prune it will remove it... I'm open to better suggestions